### PR TITLE
Multi edit option in toolbar authoring react

### PR DESCRIPTION
**Pull request by thecalcc**

### CODE REVIEW

**Comment by thecalcc** [comment]
Add a modal to choose the second article to enter multi-edit view before merging.

File: scripts/apps/authoring-react/toolbar/multi-edit-toolbar-action.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import {IArticle} from 'superdesk-api';
+import {MultiEditModal} from '../multi-edit-modal';
+import {Button, Modal, MultiSelect} from 'superdesk-ui-framework/react';
+import {Spacer} from 'core/ui/components/Spacer';
+import {showModal} from '@superdesk/common';
+import {gettext} from 'core/utils';
+import {sdApi} from 'api';
+
+interface IProps {
+    selectedArticles: Array<IArticle>;
```
Comment (tomaskikutis) [inline scripts/apps/authoring-react/toolbar/multi-edit-toolbar-action.tsx:11]: `selectedArticles` -> `initiallySelectedArticles`

File: scripts/apps/authoring-react/toolbar/multi-edit-toolbar-action.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                contentPadding="none"
+                zIndex={1050}
+                onHide={this.props.onClose}
+                visible
+                headerTemplate={gettext('Select articles')}
+                size="medium"
+            >
+                <Spacer v gap="8" noWrap style={{padding: 10}}>
+                    <MultiSelect
+                        optionLabel="slugline"
```
Comment (tomaskikutis) [inline scripts/apps/authoring-react/toolbar/multi-edit-toolbar-action.tsx:40]: use nameof - no magic strings

File: scripts/apps/authoring-react/toolbar/multi-edit-toolbar-action.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                        value={this.state.selectedArticles}
+                        onChange={(values) => {
+                            this.setState({selectedArticles: values as Array<IArticle>});
```
Comment (tomaskikutis) [inline scripts/apps/authoring-react/toolbar/multi-edit-toolbar-action.tsx:43]: remove `as` assertion - the interface was fixed in ui-framework

**Review by tomaskikutis — COMMENTED** [review]
try finding that special function for label
that should also fix bugs caused by label not being present
test again with articles that don't have a slugline neither other fields that the special function uses

**Comment by thecalcc** [comment]
> try finding that special function for label that should also fix bugs caused by label not being present test again with articles that don't have a slugline neither other fields that the special function uses

Still need to check how to properly modify `optionLabel` prop of MultiSelect in primereact so that we have a function returning a string instead of just a string. That would allow us to be more flexible in choosing what to display as label.

**Comment by thecalcc** [comment]
Changes before merging: 

https://sofab.atlassian.net/browse/SDESK-6837

Use the new optionLabel for MultiSelect component.

**Pull request merged**